metaci/build/tests/test_flows.py

```diff
@@ -198,6 +212,79 @@ def test_post_task__multiple_robot_tasks(get_spec):
         assert 3 == TestMethod.objects.all().count()
 
 
+@responses.activate
+@pytest.mark.django_db
+def test_post_task_gus_bus_test_results_enabled(get_spec, mocker, mocked_responses):
+    """Test for scenario where there are multiple Robot tasks defined
+    in a single flow. We want to make sure that test results and related
+    assets are created and associated with the correct related objects and that
+    the proper api endpoint is called."""
+    mocker.patch(
+        "metaci.build.flows.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+        METACI_RELEASE_WEBHOOK_ISSUER="MetaCI",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="test",
+        DJANGO_TIME_ZONE="US/Pacific",
+        RESULT_EXPORT_ENABLED=True,
+    )
+    mocker.patch(
+        "metaci.build.tests.test_flows.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+    )
+    mocker.patch(
+        "metaci.testresults.robot_importer.settings",
+        METACI_RELEASE_WEBHOOK_URL="https://webhook",
+        METACI_RELEASE_WEBHOOK_ISSUER="MetaCI",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="test",
+    )
+    mocker.patch(
+        "metaci.release.utils.settings",
+        METACI_RELEASE_WEBHOOK_ISSUER="https://webhook",
+        METACI_RELEASE_WEBHOOK_AUTH_KEY="12345",
+    )
+
+    with temporary_dir() as output_dir:
+        output_dir = Path(output_dir)
+
+        copyfile(
+            (TEST_ROBOT_OUTPUT_FILES / "robot_1.xml"),
+            (output_dir / "output.xml"),
+        )
+
+        step_spec = get_spec("1", name="Robot", cls=Robot)
+        step_result = StepResult(
+            step_num="1",
+            task_name="Robot",
+            path="Robot",
+            result="Pass",
+            return_values={"robot_outputdir": str(output_dir)},
+            exception=None,
+        )
+
+        responses.add(
+            "POST",
+            f"{settings.METACI_RELEASE_WEBHOOK_URL}/test-results/",
+            json={
+                "success": True,
+            },
+        )
+        build_flow = BuildFlowFactory()
+        metaci_callbacks = MetaCIFlowCallback(build_flow.id)
+        metaci_callbacks.post_task(step_spec, step_result)
+        # For output_1 we should have a single BuildFlowAsset,
+        # a single TestResult, and no TestResultAssets (screenshots)
+        assert (
+            1
+            == BuildFlowAsset.objects.filter(
+                category="robot-output", build_flow=build_flow
+            ).count()
+        )
+        assert 1 == TestResult.objects.all().count()
+        assert 0 == TestResultAsset.objects.all().count()
+        assert len(responses.calls) == 1
+        assert responses.calls[0].request.url == "https://webhook/test-results/"
```

Comment:
Should we assert something more specific about what we're sending there?

Reply:
Do you have any suggestions?